### ISSUE TITLE
Alexa developer console hangs when returning response from local debugger

### ISSUE THREAD

**Issue by jhonj98** [alexa/alexa-skills-kit-sdk-for-nodejs#660]
Hi everyone,

Please can you help me with the following issue:

I'm using ask-sdk-local-debug for debugging my skill code, I normally test it without debugging using my lambda function deployed in amazon lambda, but now I want to debug my skill code, I'm using VS Code, when I launch my skill from the developer console testing, everythings seem to run ok, even I can see my debug starting in VS Code, with the information in local console too, so far so good, the thing is that never get a response and if I see the local console log there was a response, but this never gets to the developer console. here the screenshots

![image](https://user-images.githubusercontent.com/10300948/96312549-7165a380-0fd1-11eb-97e2-87e2777ab643.png)

![image](https://user-images.githubusercontent.com/10300948/96312819-023c7f00-0fd2-11eb-9869-16d7084d2578.png)

After a few seconds I see a time_out error in the local debug console log with the error "An exception occurred while dispatching the request to the skill".

![image](https://user-images.githubusercontent.com/10300948/96312946-3f087600-0fd2-11eb-9cdf-448117c6d9ff.png)

![image](https://user-images.githubusercontent.com/10300948/96312840-0ec0d780-0fd2-11eb-9f82-90375abd0820.png)

I hope someone can help me.

Thank you in advanced.

Jhon Jairo.

**Comment by ShenChen93** [comment]
Hi @jhonj98 ,

Thanks for using our SDK and the new local debug feature. For the timeout error, it's as expected as alexa service will timeout if no response is returned after a while(If you are using breakpoint, the response will be hold and service side will report this error). But it seems there's no breakpoint hit based on your screen cut. I need to investigate why response is not send back to the service. Could you please provide us with the skill id ? 

Thanks,
Shen

**Comment by jhonj98** [comment]
Hello @ShenChen-Amazon ,

Thanks for your fast reply, the skill id is amzn1.ask.skill.2b8bf215-1cb4-4a60-adc6-ff7b4ab2bcce, the language of the skill is es_MX. And yes, there is no break point in my code.

Kind regards.

Jhon Jairo.

**Comment by sattpat** [comment]
Hi @jhonj98. Thanks for reporting this issue. Could you rerun this error scenario and notify us on this ticket as soon as you do so?

**Comment by jhonj98** [comment]
Hi @sattpat , I tested again the same scenario and the result is the same, no response yet in the developer console and timeout error in local console log.

Thank you.

Jhon Jairo.

**Comment by sattpat** [comment]
And just to make sure, if you were to stop debugging, does the skill interaction work against the configured lambda/https endpoint?

**Comment by jhonj98** [comment]
Yes! if I stop the debugging, and restart my skill testing in the developer console, the skill returns working with my configured lambda/https endpoint.

Jhon Jairo.

**Comment by lucapalano** [comment]
Same issue for me. Any news about this?

**Comment by sattpat** [comment]
Apologies for not updating the issue. Last time around I wasnt able to simulate the issue. @lucapalano and @jhonj98, is the locale you are using for testing, consistent with language used in your skill code base? In case of @jhonj98 , the response returned by your LOCAL skill code is in MX as well? @lucapalano and @jhonj98 if you can supply me the latest skill id that you are using when encountering this issue, that would be great.

**Comment by lucapalano** [comment]
Hi everyone,

> is the locale you are using for testing, consistent with language used in your skill code base?

I don't know what NodeJS version was used by @jhonj98 . I'm using NodeJS 14.x with VSCode 1.53.2 on Mac OS X. I also tested it with embedded Skill Simulation in VSCode and from ask cli.

However, I found a solution to my issue. In my case, the timeout was due to the wrong handler registration:
```
let skill;

exports.handler = async function (event, context) {
  console.log(`REQUEST++++${JSON.stringify(event)}`);
  if (!skill) {
    skill = Alexa.SkillBuilders.custom()
      .addRequestHandlers(
        LaunchRequestHandler,
        HelloWorldIntentHandler,
        HelpIntentHandler,
        CancelAndStopIntentHandler,
        SessionEndedRequestHandler,
      )
      .addErrorHandlers(ErrorHandler)
      .create();
  }

  const response = await skill.invoke(event, context);
  console.log(`RESPONSE++++${JSON.stringify(response)}`);

  return response;
};
```
It was copied from [this amazon guide](https://developer.amazon.com/en-US/docs/alexa/alexa-skills-kit-sdk-for-nodejs/develop-your-first-skill.html). It worked when deployed remotely in the lambda but doesn't work when testing locally. Here is the working code:
```
exports.handler = Alexa.SkillBuilders.custom()
      .addRequestHandlers(
        LaunchRequestHandler,
        HelloWorldIntentHandler,
        HelpIntentHandler,
        CancelAndStopIntentHandler,
        SessionEndedRequestHandler,
      )
      .addErrorHandlers(ErrorHandler)
      .lambda();
```
In this case, no timeout has been generated when tested locally.

I don't know if this will also fix the @jhonj98 issue. 

I hope this will help other people.

Best,
Luca

**Comment by sattpat** [comment]
Marking the issue for closure. Please reopen if the issue hasnt been resolved.

**Comment by arunsingh28** [comment]
.

**Comment by FelixSelter** [comment]
Same issue for me

**Comment by doiron** [comment]
Hi FelixSelter,  could you post the code where you build the skill, ie ```Alexa.SkillBuilders```  and comment on your skillId at the same time you attempt so I can know when to search for in the logs.

thanks

**Comment by techandsoftware** [comment]
I've got the same issue.  Debugging with `--region EU` was unreliable, the debug console would hang then eventually get an exception.   Today, local debugging doesn't even start with `--region EU`, as I get a websocket 500 error.   I switched to `--region NA` and downloaded the same skill with an NA account, and this debugs reliably.
